src/lib/redactData.js

```diff
@@ -0,0 +1,13 @@
+const { redactedDeathData, redactedMarriageData } = require('./redactDataModels');
+
+const redactData = (data, roles, dataset) => {
+  switch (dataset) {
+    case 'death':
+      return roles.includes("full-details") ? data : redactedDeathData(data);
+    case 'marriage':
+      return roles.includes("full-details") ? data : redactedMarriageData(data);
+    default: throw Error('Dataset not found');
+  }
+}
```

Comment:
I don't think this function serves much of a purpose. It's turning what would be a compile / linting error into a run-time error.

It would be better to have `redactDeath`, `redactMarriage` etc. as you did before.

Reply:
If you really wanted to be clever and avoid repeating the logic around `full-details`, you could pass in the function.
```
const redactData = redact => (data, roles) => (
  roles.includes('full-details')
    ? data
    : redact(data)
);

const redactDeath = redactData(redactedDeathData);
const redactMarriage = redactData(redactedMarriageData);
```